src/logic/Applications.swift

```diff
@@ -32,10 +40,11 @@ class Applications {
             let windowsOnOtherSpaces = Spaces.windowsInSpaces(otherSpaces)
             let windowsOnlyOnOtherSpaces = Array(Set(windowsOnOtherSpaces).subtracting(windowsOnCurrentSpace))
             if windowsOnlyOnOtherSpaces.count > 0 {
-                // on initial launch, we use private APIs to bring windows from other spaces into the current space, observe them, then remove them from the current space
-                CGSAddWindowsToSpaces(cgsMainConnectionId, windowsOnlyOnOtherSpaces as NSArray, [Spaces.currentSpaceId])
-                Applications.observeNewWindowsBlocking()
-                CGSRemoveWindowsFromSpaces(cgsMainConnectionId, windowsOnlyOnOtherSpaces as NSArray, [Spaces.currentSpaceId])
+                // Currently we add those window in other space without AXUIElement init
+                // We don't need to get the AXUIElement until we focus these windows.
```

Comment:
If you're correct that `CGSMoveWindowsToManagedSpace` is a perfect replacement to `CGSAddWindowsToSpaces`, then we could at least fix the 12.3 issue with `CGSAddWindowsToSpaces` not working, but replacing the call to `CGSMoveWindowsToManagedSpace`. That would fix the issue. Then, separately, we can have the conversation about the invisible window trick, and how to move forward there

Reply:
Please see my follow-up here: https://github.com/lwouis/alt-tab-macos/issues/1324#issuecomment-1120156153